### PR TITLE
Math: Fixes and cleanup to Box and Size

### PR DESCRIPTION
**Pull request by marcusnaslund**
Replaces #610 because VSCode messed up my git history (at least that's what I'm blaming it on).

### CODE REVIEW

File: source/math/FloatBox2D.ooc [hunk 1]
```diff
@@ -159,18 +159,18 @@ FloatBox2D: cover {
 		array free()
 		result
 	}
-	create: static func (leftTop: FloatPoint2D, size: FloatSize2D) -> This { This new(leftTop, size) }
-	create: static func ~fromFloats (left, top, width, height: Float) -> This { This new(left, top, width, height) }
-	createAround: static func (center: FloatPoint2D, size: FloatSize2D) -> This { This new(center + (-size) / 2, size) }
+	createAround: static func (center: FloatPoint2D, size: FloatSize2D) -> This { This new(center - (size / 2.0f), size) }
```
Comment (emilwestergren) [inline source/math/FloatBox2D.ooc]: Parenthesises are unnecessary:
`center - size / 2.0f`

File: source/math/IntBox2D.ooc [hunk 2]
```diff
@@ -128,18 +129,18 @@ IntBox2D: cover {
 		array free()
 		result
 	}
-	create: static func (leftTop: IntPoint2D, size: IntSize2D) -> This { This new(leftTop, size) }
-	create: static func ~fromFloats (left, top, width, height: Int) -> This { This new(left, top, width, height) }
-	createAround: static func (center: IntPoint2D, size: IntSize2D) -> This { This new(center + (-size) / 2, size) }
-	bounds: func (left, right, top, bottom: Int) -> This { This new(left, top, right - left, bottom - top) }
-	bounds: func ~fromArray (points: IntPoint2D[]) -> This { this bounds(points as ArrayList<IntPoint2D>) }
-	bounds: func ~fromList (points: ArrayList<IntPoint2D>) -> This {
+	createAround: static func (center: IntPoint2D, size: IntSize2D) -> This { This new(center - (size / 2), size) }
```
Comment (emilwestergren) [inline source/math/IntBox2D.ooc]: Parentheses are unnecessary here:
`center - size / 2`

Comment (marcusnaslund) [inline source/math/IntBox2D.ooc]: Thanks. Fixed.

File: source/math/FloatBox2D.ooc [hunk 3]
```diff
@@ -34,7 +34,7 @@ FloatBox2D: cover {
 	rightTop ::= FloatPoint2D new(this right, this top)
 	leftBottom ::= FloatPoint2D new(this left, this bottom)
 	rightBottom ::= this leftTop + this size
-	center ::= this leftTop + (this size / 2)
+	center ::= this leftTop + (this size / 2.0f)
```
Comment (emilwestergren) [inline source/math/FloatBox2D.ooc]: Parentheses unnecessary

File: test/math/FloatBox2DTest.ooc [hunk 4]
```diff
@@ -23,19 +25,38 @@ FloatBox2DTest: class extends Fixture {
...
-			expect(size height, is equal to(4.0f))
+			expect(size width, is equal to(3.0f) within (this precision))
+			expect(size height, is equal to(4.0f) within (this precision))
+		})
+		this add("addition, union", func {
+			result := box0 + box1
+			expect(result left, is equal to(1.0f) within (this precision))
+			expect(result top, is equal to(2.0f) within (this precision))
+			expect(result width, is equal to(5.0f) within (this precision))
+			expect(result height, is equal to(4.0f) within (this precision))
```
Comment (niklasborwell) [inline test/math/FloatBox2DTest.ooc]: Shouldn't be a space after within, I would assume, since it breaks the color coding.

Comment (marcusnaslund) [inline test/math/FloatBox2DTest.ooc]: I see nothing wrong with the colours? Anyway, you're right.

File: source/math/FloatSize2D.ooc [hunk 5]
```diff
@@ -33,9 +33,9 @@ FloatSize2D: cover {
 	init: func@ ~square (length: Float) { this width = this height = length }
 	init: func@ ~default { this init(0.0f, 0.0f) }
 	pNorm: func (p: Float) -> Float {
-		p == 1 ?
-		this width abs() + this height abs() :
-		(this width abs() pow(p) + this height abs() pow(p)) pow(1 / p)
+		p == 1.0f ?
```
Comment (emilwestergren) [inline source/math/FloatSize2D.ooc]: Don't do equal comparisons with floats, compare the difference with absolute or relative error margin.

File: source/math/FloatSize3D.ooc [hunk 6]
```diff
@@ -33,9 +33,9 @@ FloatSize3D: cover {
 	init: func@ (=width, =height, =depth)
 	init: func@ ~default { this init(0.0f, 0.0f, 0.0f) }
 	pNorm: func (p: Float) -> Float {
-		p == 1 ?
-		this width abs() + this height abs() :
-		(this width abs() pow(p) + this height abs() pow(p)) pow(1 / p)
+		p == 1.0f ?
```
Comment (emilwestergren) [inline source/math/FloatSize3D.ooc]: Don't do equal comparisons with floats, compare the difference with absolute or relative error margin.

File: source/math/FloatSize2D.ooc (hunk 5 above)
Comment (marcusnaslund) [inline source/math/FloatSize2D.ooc]: Should we even keep the special case at all?

Comment (niklasborwell) [inline source/math/FloatSize2D.ooc]: I don't think it's necessary.

Comment (emilwestergren) [inline source/math/FloatSize2D.ooc]: Remove it

**Comment by davidhesselbom** [comment]
Waiting for @emilwestergren or @niklasborwell to mark this as peer reviewed.

**Comment by marcusnaslund** [comment]
Done.

**Pull request merged**